### PR TITLE
Github to GitHub

### PR DESCRIPTION
**Pull request by midnightconman**
In working with various functions and methods over the last two weeks, I noticed that some functions use `GitHub` and some use `Github`, it appears that `GitHub` is the winner for popularity right now:

*on master (e126eb0)*
Github: 360
GitHub: 1196

I think it would be nice to avoid confusion for types, functions, etc.

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
If we want to land something like this can we:

 - be upfront about how the changes were generated (is this a `sed`?)
 - make the diff reproducible by someone else with those commands
 - add a presubmit validation to disallow Github -- can we add that to our spell checker?

**Comment by midnightconman** [comment]
> If we want to land something like this can we:
> 
> * be upfront about how the changes were generated (is this a `sed`?)

This makes sense... I used the following `sed` command to produce this PR:
```
$ find . -name '*.go' -type f -exec sed -i -e 's/Github/GitHub/g' {} \;
```
But it should be improved to exclude the `vendors` directory, like so?
```
$ find . \( -name '*.go' -not -path '*vendor*' \) -type f -exec sed -i -e 's/Github/GitHub/g' {} \;
```

> * make the diff reproducible by someone else with those commands

How should this be submitted? As a comment, or possibly another PR with the same diff?

> * add a presubmit validation to disallow Github -- can we add that to our spell checker?

That sounds good... let me see where that is located and push a commit.

**Comment by stevekuznetsov** [comment]
As long as you provide a shell script that I can copy-paste into my editor and validate your commits that is fine.

**Comment by midnightconman** [comment]
> * add a presubmit validation to disallow Github -- can we add that to our spell checker?

I found the `kubernetes/test-infra` presubmit jobs here: https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes/test-infra

And I found `misspell`: https://github.com/kubernetes/test-infra/blob/master/hack/verify-spelling.sh

Can I add custom strings to `misspell`? Or should I create something custom to check for instances of `Github` in submitted files?

**Comment by stevekuznetsov** [comment]
I'm not sure -- @BenTheElder @ixdy do you know?

**Comment by BenTheElder** [comment]
`misspell` has a `-i` ignore flag for a CSV of misspellings to ignore.

**Comment by BenTheElder** [comment]
er, but it does not seem to be customizable to add additional ones

**Comment by stevekuznetsov** [comment]
:man_facepalming: 

Let's just add a `grep -nr`

**Comment by midnightconman** [comment]
Does this work?
```
grep -rn 'Github' /workspace && echo -en '\n-- Please change all instances of Github to GitHub --\n' ; exit 1
```

**Comment by stevekuznetsov** [comment]
Does that exit zero after your changes? Not sure if the `&&` expression works on the next command or the rest of the line. Maybe

```sh
if grep -rq 'Github' /workspace; then 
    echo -en '\n-- Please change all instances of Github to GitHub --\n'
    exit 1
fi
```

**Comment by midnightconman** [comment]
Yep, they are equivalent. The way this reads is...
If grep - rn 'Github' exits 0, then echo...
If echo... exits 0, then exit 1

It's just shorthand Bourne. Writing an if statement on one line would be something like:
```
if ( grep - rn 'Github' /workspace ) ; then ; echo -en... ; exit 1 ; fi
```

**Comment by midnightconman** [comment]
Let me check in the script in /hack, so we don't need the weird oneliner

**Comment by midnightconman** [comment]
Okie dokie, @stevekuznetsov - Please look again.

I added the one liner as a script in `/hack` and fixed the presubmit to use the script.

File: hack/verify-github-spelling.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+# Copyright 2017 The Kubernetes Authors.
```
Comment (stevekuznetsov) [inline hack/verify-github-spelling.sh]: 2019

**Comment by midnightconman** [comment]
After looking at testing results, I came up with a question...

The find and replace above is only changing `.go` files, do we only want this to check go files, or other files? At the moment the verify-spelling script is checking all files. If we only want to check `.go` files, I need to update the grep to match the find and replace. :smile:

**Comment by stevekuznetsov** [comment]
I think all project files (not vendor) would be good, but it's not super important to me -- what do you think?

**Comment by midnightconman** [comment]
Sounds good to me... I used the following script to include all files and exclude `.git` and `vendor`:
```
find . \( -not -path '*vendor*' -and -not -path '*.git*' \) -type f -exec sed -i -e 's/Github/GitHub/g' {} \;
```

Also updated the script to match this logic.

**Comment by midnightconman** [comment]
@stevekuznetsov - Please take another look.

Let me know and I'll squash these commits.

File: hack/verify-github-spelling.sh [hunk 2]
```diff
@@ -13,12 +13,15 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-set -o errexit
```
Comment (stevekuznetsov) [inline hack/verify-github-spelling.sh]: Definitely want to keep these

File: gubernator/github/main_test.py [hunk 3]
```diff
@@ -60,28 +60,28 @@ def get_response(self, event, body):
             body = json.dumps(body)
         signature = handlers.make_signature(body)
         resp = app.post('/webhook', body,
-            {'X-Github-Event': event,
```
Comment (midnightconman) [inline gubernator/github/main_test.py]: Can I have someone verify this won't be an issue? I think case doesn't matter, but I want to verify that.

The spec shows the _corrected_ form: https://developer.github.com/webhooks/#delivery-headers

File: hack/verify-github-spelling.sh [hunk 4]
```diff
@@ -13,12 +13,15 @@
...
 set -o nounset
 set -o pipefail
 
-if grep -rn 'Github' /workspace
+find . \( -not -path '*vendor*' -and -not -path '*.git*' -and -not -path '*hack/verify-github-spelling.sh*' \) -type f -exec grep -Hn 'Github' {} +
+
+if [ $? -eq 0 ]
```
Comment (stevekuznetsov) [inline hack/verify-github-spelling.sh]: So we generally don't allow filenames with spaces etc (so no need for `-print0` and no issued with the `for` loop),  so you could do

```sh
failed="false"
for file in $(  find . \( -not -path '*vendor*' -and -not -path '*.git*' -and -not -path '*hack/verify-github-spelling.sh*' \) -type f ); do
    if grep -q 'Github' "${file}"; then
        echo "[ERROR] ${file}: contains mis-spelling 'Github'" 
        failed="true"
    fi
done

if [[ "${failed}" == "true" ]]; then
    exit 1
fi
```

Your discretion -- but I feel like writing Bash that looks like real languages is easier to understand and maintain. @BenTheElder likely has opinions too

File: hack/verify-github-spelling.sh (hunk 1 above)
Comment (midnightconman) [reply hack/verify-github-spelling.sh]: Corrected :smile:

File: hack/verify-github-spelling.sh (hunk 4 above)
Comment (midnightconman) [reply hack/verify-github-spelling.sh]: Looks good to me... updated the script.

File: hack/verify-github-spelling.sh (hunk 2 above)
Comment (midnightconman) [reply hack/verify-github-spelling.sh]: Added back in update.

File: gubernator/github/main_test.py (hunk 3 above)
Comment (stevekuznetsov) [reply gubernator/github/main_test.py]: LOL

Comment (stevekuznetsov) [reply gubernator/github/main_test.py]: Let's run the tests and see?

**Comment by midnightconman** [comment]
This PR is missing `ok-to-test` :disappointed:

**Comment by stevekuznetsov** [comment]
LOL what?

**Comment by stevekuznetsov** [comment]
Wow, we never even did it I guess

/ok-to-test

File: config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml [hunk 5]
```diff
@@ -268,3 +268,16 @@ presubmits:
...
+    - master
+    always_run: true
+    decorate: true
+    labels:
+      preset-service-account: "true"
+    spec:
+      containers:
+      - image: gcr.io/k8s-testimages/kubekins-e2e:v20190220-615bdbe2e-experimental
+        command:
+        - ./hack/verify-github-spelling.sh
```
Comment (BenTheElder) [inline config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml]: job config looks sane :+1:

**Comment by midnightconman** [comment]
Sorry about the force-pushing... I wanted to rebase master on more time.

**Comment by midnightconman** [comment]
Ohh, I don't know what this means:
```
pull-test-infra-verify-github-spelling does not have a matching testgrid testgroup
```

**Comment by midnightconman** [comment]
@stevekuznetsov @BenTheElder - Please take another look... All Tests Passed! :smile:

File: prow/plank/controller.go [hunk 6]
```diff
@@ -165,7 +165,7 @@ func (c *Controller) setPreviousReportState(pj prowapi.ProwJob) error {
 		latestPJ.Status.PrevReportStates = map[string]prowapi.ProwJobState{}
 	}
 	latestPJ.Status.PrevReportStates[reporter.GitHubReporterName] = latestPJ.Status.State
-	_, err = c.prowJobClient.Update(latestPJ)
+	_, err = c.kc.ReplaceProwJob(latestPJ.ObjectMeta.Name, latestPJ)
```
Comment (BenTheElder) [inline prow/plank/controller.go]: what is this about?

Comment (midnightconman) [reply prow/plank/controller.go]: Sorry about that... I think I accidentally pulled in a change during rebase. When I squash that will go away.

**Comment by stevekuznetsov** [comment]
Instead of rebasing, it's best for these to re-run your script to generate the changes -- that's why I suggested we be explicit about how to generate the changes, so it's easier to review and harder to make mistakes

**Comment by midnightconman** [comment]
That makes sense :smile:

**Comment by midnightconman** [comment]
@stevekuznetsov @BenTheElder - can you guys please take one last look at this? Thank you! :smile:

**Comment by krzyzacy** [comment]
I feel like we have too many always_run presubmit jobs in test-infra now...

**Comment by krzyzacy** [comment]
(raise the concern since we have the # pod limit)

**Comment by stevekuznetsov** [comment]
What is wrong with more presubmits? The more granular a presubmit is, the faster it is to re-test if it fails or flakes ... what is a # pod limit? Why do you have one?

> edit: also it breaks HEAD...

:(

> also - we don't run linting as a presubmit, if we want to go this route, we might as well golint all affected go files?

We should discuss it at the meeting today

**Comment by krzyzacy** [comment]
> Why do you have one?

Unfortunately, but see https://github.com/kubernetes/test-infra/issues/11594#issuecomment-469484840

**Pull request merged**